src/control/content/components/LocationsActionBar.js

```diff
@@ -70,14 +70,20 @@ class LocationsActionBar extends React.Component {
       // otherwise, wait for complete and merge
       // the results
       if (!promises.length) {
-        this.props.onMultipleSubmit(locations);
+        while(locations.length){
```

Comment:
Why do we have a sync and async operations here?

Reply:
There is a comment above in the code that says
"if a row is missing latitude or longitude use google maps api to fetch them async otherwise just return the location"
First every row is checked whether it has lat / lng. If row without latlng is found, they add request to geocode it to aray.
Once every row is checked, if there are rows to be geocoded, Promise.all is used to geocode them and then data is sent to server. In case there is no row that needs to be geocoded, data is sent to server directly without waiting for google service to finish geocoding.

Reply:
great, thanks